plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+                await self.schedule_prompt()
+
+    async def process_message(self, message):
+        """
+        Process a non-command message from the moderator.
+        If there is a current active ticket, treat it as a comment.
+        Either way, update the last handled message in data.
+        """
+        if not priv.has_privilege('mod', message.author):
+            # Don't process messages from non-moderators at all.
+            return
+
+        if not commands.conf.prefix or not message.content.startswith(commands.conf.prefix):
+            content = message.content
+            if ticket := self.current_ticket:
+                logger.info(
+                    "Processing message from moderator {} as comment to ticket #{}: {}".format(self.modid,
+                                                                                               ticket.id,
+                                                                                               repr(content))
+                )
+
+                # Parse the message as a comment to the current ticket
+                if match := ticket_comment_re.match(content):
+                    # Extract duration
+                    if match[1]:
+                        d = int(match[1])
+                        token = match[2][0]
+                        token = token.lower() if token != 'M' else token
+                        duration = d * time_expansion[token]
+                    else:
+                        duration = None
+
+                    # Extract comment
+                    comment = content[match.end():]
+                else:
+                    duration = None
+                    comment = content
+
+                # Update the ticket
+                with ticket.batch_update():
+                    ticket.stage = TicketStage.COMMENTED
+                    ticket.duration = duration if (ticket.can_revert and ticket.active) else None
+                    ticket.comment = comment
+                    ticket.modified_by = self.modid
+                    if ticket.status == TicketStatus.NEW:
+                        ticket.status = TicketStatus.IN_EFFECT
+
+                self.last_read_msgid = message.id
+
+                # Notify the moderator, nullify the duration if required
+                if duration:
+                    if not ticket.can_revert:
+                        msg = ("Ticket comment set! "
+                               "Provided duration ignored since this ticket type cannot expire.")
+                        duration = None
+                    elif not ticket.active:
+                        msg = ("Ticket comment set! "
+                               "Provided duration ignored since this ticket is no longer in effect.")
+                        duration = None
+                    else:
+                        msg = "Ticket comment and duration set!"
+                else:
+                    msg = "Ticket comment set!"
```

Comment:
I was thinking we reply with the duration, to indicate how it was parsed just in case.